RFCs/FS-1128-better-interpolated-triple-quoted-strings.md

```diff
@@ -0,0 +1,119 @@
+
+# Alternatives
+
+Alternatives:
+1. Try to fully align with C#'s raw string literals at the cost of backward compatiblity.
+2. Introduce alternative syntax for delimiting expression fills that does not clash with `{` and `}` which are common in JSON and certain other contexts.
+
+# Compatibility
+
+Please address all necessary compatibility questions:
+
+* Is this a breaking change?
+*No*
+* What happens when previous versions of the F# compiler encounter this design addition as source code?
+*A compiler error*
+* What happens when previous versions of the F# compiler encounter this design addition in compiled binaries?
+*AFAICT it should work, changes will hopefully be limited to lexing stage*
+* If this is a change or extension to FSharp.Core, what happens when previous versions of the F# compiler encounter this construct?
+*It is not a change/extension to FSharp.Core*
+
+# Pragmatics
+
+## Diagnostics
+
+* New compiler error for interpolated string literals with too many consecutive `{` or `}` characters.
+* Otherwise, same compiler errors as for regular (single-`$`) interpolated strings.
+
+## Tooling
+
+Compiler tooling should offer the same support for the extended form of interpolated string literals as it currently does for interpolated triple quoted literals. This includes colorization, autocompletion, and navigational features in an interpolated context.
+
+## Performance
+
+AFAICT there should be no impact on performance (although there might be if using certain constructs in regexes within .fsl files can have a big impact `XXX TODO`)
+
+## Scaling
+
+Please list the dimensions that describe the inputs for this new feature, e.g. "number of widgets" etc.  For each, estimate a reasonable upper bound for the expected size in human-written code and machine-generated code that the compiler will accept.
+
+For example
+
+* Expected maximum number of widgets in reasonable hand-written code: 100
+* Expected reasonable upper bound for number of widgets accepted: 500
+
+Testing should particularly check that compilation is linear (or log-linear or similar) along these dimensions.  If quadratic or worse this should ideally be noted in the RFC.
+
+`TODO` *No clue really, but I expect more than 3x$ will be very rarely used though*
+
+## Culture-aware formatting/parsing
+
+Does the proposed RFC interact with culture-aware formatting and parsing of numbers, dates and currencies? For example, if the RFC includes plaintext outputs, are these outputs specified to be culture-invariant or current-culture.
+
+*No* `TODO: remove this section?`
+
+# Unresolved questions
+
+1. Do we want to also provide a way to easily input procent signs in interpolated string literals?
```

Comment:
I think it’s “percent”, not “procent”? Though I don’t have an answer to your question ;). Seems nontrivial to support.

Reply:
I am afraid you are right about it being nontrivial. At least it is not a natural extension of changes that are needed for the main part of this RFC and it seems like it wouldn't be limited to lexing either.